src/example_documentation.cpp

```diff
@@ -0,0 +1,128 @@
+// DESCRIPTION: Appropriate license header at the top, e.g.,
+/*
+* Copyright (c) 2018, NVIDIA CORPORATION.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+
+// DESCRIPTION: A brief description of the overall purpose and contents of this file
+/** ---------------------------------------------------------------------------*
+* @brief Example code documentation for libgdf.
+*
+* This file provides examples of how source files, classes, functions, and variables
+* should be documented in libgdf.
+* ---------------------------------------------------------------------------**/
+
+
+// DESCRIPTION: A brief description of the purpose and functionality of the class 
+/* --------------------------------------------------------------------------*/
+/** 
+ * @brief  This class serves as an example of how classes in libgdf should 
+ * be documented.
+ */
+/* ----------------------------------------------------------------------------*/
+template <typename T>
+class example_class
+{
+
+  // DESCRIPTION: Trivial class functions should be given names that make their purpose clear
+  // DESCRIPTION: If their name makes the functionality obvious, no documentation is necessary
+  void get_my_int() {
+    return my_int;
+  }
+  void set_my_int(int new_value){
+    my_int = new_value;
+  }
+
+  // DESCRIPTION: Non-trivial member functions should have a brief description of the
+  // function as well as all of its parameters. Every parameter should be decorated to 
+  // indicate if it is an input or output parameter, or both with @Param[in], @Param[out],
+  // and @Param[in,out] respectively.
+  /* --------------------------------------------------------------------------*/
+  /** 
+   * @brief This is a complicated function that requires more detailed documentation.
+   * 
+   * @Param[in] first_parameter  This parameter is an input parameter to the function
+   * @Param[in,out] second_parameter This parameter is used both as an input and output
+   * @Param[out] third_parameter This parameter is an output of the function
+   * 
+   * @Returns The result of the complex function
+   */
+  /* ----------------------------------------------------------------------------*/
+  T complicated_function(int const first_parameter,
+                         double * second_parameter,
+                         float * third_parameter)
+  {
+    // DESCRIPTION: Notice the use of *human readable* variable names. Human readable 
```

Comment:
Are you suggesting a coding standard for names? E.g. python style `_` between words?

Reply:
The intent here was not to proscribe camel case vs underscores, but rather to emphasize the importance of using human readable variables, e.g., `firstParameter` or `first_parameter`, doesn't matter to me, both are infinitely preferable to `fp`. Good variable names are self-documenting.